lxd/migrate.go

```diff
@@ -292,12 +292,16 @@ func (s *migrationSourceWs) Do(migrateOp *operation) error {
 	criuType := CRIUType_CRIU_RSYNC.Enum()
 	if !s.live {
 		criuType = nil
+	}
 
-		err := s.container.StorageStart()
-		if err != nil {
-			return err
-		}
+	// COMMENT(brauner): Storage needs to start unconditionally now, since
+	// we need to initialize a new storage interface.
+	err := s.container.StorageStart()
+	if err != nil {
+		return err
+	}
```

Comment:
Yeah, this has something to do with handling container mounts. Essentially, if we didn't mount the container exclusively for us, then we shouldn't unmount it and that's why the `StorageStop()` is missing otherwise migration will fail. (The low-level implementations for `ContainerMount()` and ` ContainerUmount()` return a boolean indicating whether they actually mounted the storage volume or  only detected that it was already mounted. The `StorageStart()`/`StorageStop()` routines don't and I don't want to change them right now.) But I can handle this smarter by checking prior to `StorageStart()` If the container is mounted and then conditionalize the `StorageStop()` call.